app/docs/md/patterns/rendering-markdown.md

```diff
@@ -0,0 +1,249 @@
+   json: { doc }
+ }
+}
+```
+
+</doc-code>
+
+> **Note:** We’re omitting a lot of error checking for the sake of brevity, but check out this [hardened example](https://github.com/enhance-dev/enhance.dev/blob/main/app/api/docs/%24%24.mjs) for more details on how to handle 404s.
+
+## Displaying Markdown
+
+Now that we've successfully transformed our Markdown into HTML, let's go about displaying it in a page. First, we'll create a new catch-all page under `app/pages/markdown/$$.html`, and a new web component to display the Markdown at `app/elements/doc-content.mjs`. You can create these files manually, or by running the Enhance CLI commands:
+
+```bash
+enhance gen page --path 'markdown/$$'
+enhance gen element --name doc-content
+```
+
+Our project structure now looks like this:
+
+```
+app
+├── api
+|   └── markdown
+|       └── $$.mjs
+├── elements
+|   └── doc-content.mjs
+├── markdown
+|   └── example.md
+└── pages
+    ├── markdown
+    |   └── $$.mjs
+    └── index.html
+```
+
+For the purposes of this example, we'll keep our `app/pages/markdown/$$.html` simple by including only the web component we just created:
+
+<doc-code filename="app/pages/markdown/$$.html">
+
+```html
+<doc-content></doc-content>
+```
+
+</doc-code>
+
+In the web component we just created, we'll read the result of the transformation off the `store`. Then we'll return the rendered content to be displayed on our page:
+
+<doc-code filename="app/elements/doc-content.mjs">
+
+```javascript
+export default function DocContent ({ html, state }) {
+ const { store } = state
+ const { doc } = store
+ return html`
+<h1 class="text3">${doc.title}</h1>
+<div>
+ ${doc.html}
+</div>
+`
+}
+```
+
+</doc-code>
+
+Test it out by starting the development server:
+
+```bash
+enhance dev
+```
+
+Then open a browser tab to [localhost:3333/markdown/example](https://localhost:3333/markdown/example) and you’ll see the rendered markdown file.
+
+That's all you need in order to get started using markdown in an Enhance app.
+
+## Using custom elements in markdown
+
+It is totally possible to include custom elements in Markdown source files and then have the generated markup rendered by Enhance SSR.
+
+When authoring Markdown with custom elements, it is best to add blank lines around opening and closing tags. For example on this page we use the `doc-code` custom element to provide syntax highlighting of source code. In markdown, use of a custom element would look like:
+
+```
+# Custom Elements in Markdown
+
+Custom HTML elements in Markdown are awesome!
+
+<my-rad-elem hype="9001">
+
+## Some really cool info
+
+> This is rendered as `<h2>` and `<blockquote>` inside `<my-rad-elem>`
+
+</my-rad-elem>
+
+Hey, that's pretty neat!
+```
+### Render order matters
+
+When rendering HTML from Markdown that includes custom elements the rendering order matters. First, render the markdown to HTML with Arcdown then "enhance" that HTML by running it through Enhance SSR before sending it to the browser.
+
```

Comment:
I feel it would be good to add a few words as to why.
Something like:
since Enhance expands custom elements that would not be present in the markup until after markdown converts the source into HTML.

Reply:
Yes that is true as well

Reply:
Maybe we should just say that Enhance produces a full HTML page and not a HTML snippet like Arcdown does.

Reply:
Removed this section after chatting about it in Enhance sync